<a href="https://colab.research.google.com/github/bema1015/ProjDataEng/blob/main/Projet2/projet2_dev.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Contexte

Mark, votre manager, vous a convié à une réunion pour vous présenter le
projet d’expansion à l’international de l’entreprise. Il vous confie une
première mission d’analyse exploratoire, pour déterminer si les données sur
l’éducation de la banque mondiale permettent d’informer le projet
d’expansion.
Voici les différentes questions que Mark aimerait explorer, que vous avez
notées durant la réunion :
*   Quels sont les pays avec un fort potentiel de clients pour nos services ?
*   Pour chacun de ces pays, quelle sera l’évolution de ce potentiel de clients ?
*   Dans quels pays l'entreprise doit-elle opérer en priorité ?

# Jeu de donnée

Issu de la Banque Mondiale :
https://datacatalog.worldbank.org/dataset/education-statistics
Disponible en telchargment direct à https://s3-eu-west-1.amazonaws.com/static.ocstatic.com/prod/courses/files/Parcours_data_scientist/Projet+-
+Donn%C3%A9es+%C3%A9ducatives/Projet+Python_Dataset_Edstats_csv.zip
Le fichier zip contient 5 fichiers csv :
* EdStatsData.csv
* EdStatsFootNote.csv
* EdStatsSeries.csv
* EdStatCountry.csv
* EdStatsCountry-Series.csv


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# 1°/ Chargement & Aperçu

Objectif :
* Charger les données

Résumé :
* EdStatsData.csv = EDSDATA
* EdStatsFootNote.csv = EDSFootNote
* EdStatsSeries.csv = EDSSeries
* EdStatCountry.csv = EDSCountry
* EdStatsCountry-Series.csv = EDSCountrySeries

In [3]:
EDSData = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Projet+Python_Dataset_Edstats_csv/EdStatsData.csv')
EDSData.head()

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

In [4]:
EDSCountrySeries = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Projet+Python_Dataset_Edstats_csv/EdStatsCountry-Series.csv')
EDSCountrySeries.head()

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

In [5]:
EDSCountry = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Projet+Python_Dataset_Edstats_csv/EdStatsCountry.csv')
EDSCountry.head()

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

In [6]:
EDSFootNote = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Projet+Python_Dataset_Edstats_csv/EdStatsFootNote.csv')
EDSFootNote.head()

CountryCode      SeriesCode    Year          DESCRIPTION  Unnamed: 4
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.         NaN
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.         NaN
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.         NaN
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.         NaN
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.         NaN

In [7]:
EDSSeries = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Projet+Python_Dataset_Edstats_csv/EdStatsSeries.csv')
EDSSeries.head()
#

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition  Unit of measure  \
0  Percentage of female population age 15-19 with...              NaN   
1  Percentage of population age 15-19 with no edu...              NaN   
2  Percentage of female population age 15+ with n...              NaN   
3  Percentage of population age 15+ with no educa...              NaN   
4  Percentage of female population age 20-24 with...              NaN   

  Periodicity Base Period Other notes Aggregation method  ...  \
0         NaN         NaN         NaN                NaN  ...   
1         NaN         NaN         NaN                NaN  ...   
2         NaN         NaN         NaN                NaN  ...   
3         NaN         NaN         NaN                NaN  ...   
4         NaN         NaN         NaN                NaN  ...   

  Notes from original source  General comments  \
0                        NaN               NaN   
1                        NaN               NaN   
2                        NaN               NaN   
3                        NaN               NaN   
4                        NaN               NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links Other web links  Related indicators  License Type  \
0                  NaN             NaN                 NaN           NaN   
1                  NaN             NaN                 NaN           NaN   
2                  NaN             NaN                 NaN           NaN   
3                  NaN             NaN                 NaN           NaN   
4                  NaN             NaN                 NaN           NaN   

   Unnamed: 20  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  

[5 rows x 21 columns]

# 2°/ Qualité des données

Objectifs :
* Nettoyer les données

#EDSDATA

In [ ]:
EDSData.tail()

Country Name Country Code  \
886925     Zimbabwe          ZWE   
886926     Zimbabwe          ZWE   
886927     Zimbabwe          ZWE   
886928     Zimbabwe          ZWE   
886929     Zimbabwe          ZWE   

                                           Indicator Name  \
886925  Youth illiterate population, 15-24 years, male...   
886926  Youth literacy rate, population 15-24 years, b...   
886927  Youth literacy rate, population 15-24 years, f...   
886928  Youth literacy rate, population 15-24 years, g...   
886929  Youth literacy rate, population 15-24 years, m...   

              Indicator Code  1970  1971  1972  1973  1974  1975  ...  2060  \
886925      UIS.LP.AG15T24.M   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886926     SE.ADT.1524.LT.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886927  SE.ADT.1524.LT.FE.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886928  SE.ADT.1524.LT.FM.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886929  SE.ADT.1524.LT.MA.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   

        2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
886925   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886926   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886927   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886928   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886929   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

==> Les 4 premieres colonne contiennent des informations qualitatif (Nom du pays, code du pays, nom de l'indicateur, code de l'indicateur). Puis les autres correpondent au valeur associé en fonction des années. La dernière est nommé Unnamed : 69 (sens inconnu). 886930 lignes pour 70 colonnes

1°/ Erreurs de type

In [ ]:
EDSData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

Rien ne semble est problèmatique avec les types, par contre la colonne Unnamed : 69 ne correspond à rien

In [ ]:
EDSData_copy = EDSData.copy()

In [ ]:
EDSData = EDSData.drop(columns=["Unnamed: 69"])
EDSData.head()

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2055  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2060  2065  2070  2075  2080  2085  2090  2095  2100  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[5 rows x 69 columns]

2°/ Valeurs manquantes

In [ ]:
EDSData.isnull().sum()

Country Name           0
Country Code           0
Indicator Name         0
Indicator Code         0
1970              814642
                   ...  
2080              835494
2085              835494
2090              835494
2095              835494
2100              835494
Length: 69, dtype: int64

In [ ]:
pd.set_option("display.max_columns", None)
missing_percent = EDSData.isnull().mean()*100

pd.set_option("display.max_rows", None)
missing_percent.sort_values(ascending=False)

2017              99.983877
2016              98.144160
1971              95.993258
1973              95.992356
1972              95.984012
1974              95.971497
1979              95.849842
1976              95.773849
1982              95.770692
1989              95.767422
1977              95.763589
1978              95.763364
1983              95.663694
1988              95.653321
1984              95.647233
1987              95.643286
1981              95.627953
1986              95.560867
2045              94.200670
2100              94.200670
2030              94.200670
2095              94.200670
2090              94.200670
2025              94.200670
2080              94.200670
2085              94.200670
2050              94.200670
2035              94.200670
2040              94.200670
2065              94.200670
2070              94.200670
2075              94.200670
2060              94.200670
2055              94.200670
2020              94.200670
1970              91.849639
1997              91.718287
1991              91.607342
1992              91.482642
1993              91.454455
1996              91.340128
1994              91.266278
1998              90.426076
1975              90.156382
1980              89.951631
1985              89.819264
2014              87.170464
1999              86.601085
2001              86.074549
2002              85.996076
1990              85.973527
2004              85.476419
2003              85.301771
2015              85.223411
1995              85.189248
2008              84.848071
2007              84.522792
2013              84.496071
2006              84.180037
2009              83.977541
2011              83.537370
2012              83.396209
2000              80.080051
2005              79.242105
2010              72.665036
Indicator Code     0.000000
Indicator Name     0.000000
Country Code       0.000000
Country Name       0.000000
dtype: float64

Objectif : supprimer les lignes qui sont toutes en NaN

In [ ]:
# Sélection des colonnes 4 à 67
cols = EDSData.columns[4:68]

# Supprimer lignes où toutes ces colonnes sont NaN
EDSData_clean = EDSData.dropna(subset=cols, how="all")
EDSData_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 357405 entries, 4 to 886929
Data columns (total 69 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    357405 non-null  object 
 1   Country Code    357405 non-null  object 
 2   Indicator Name  357405 non-null  object 
 3   Indicator Code  357405 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983           

In [ ]:
missing_percent_clean = EDSData_clean.isnull().mean()*100

pd.set_option("display.max_rows", None)
missing_percent_clean.sort_values(ascending=False)

2017              99.959989
2016              95.394580
1971              90.056938
1973              90.054700
1972              90.033995
1974              90.002938
1979              89.701039
1976              89.512458
1982              89.504624
1989              89.496510
1977              89.486997
1978              89.486437
1983              89.239099
1988              89.213357
1984              89.198248
1987              89.188456
1981              89.150404
1986              88.983926
2045              85.608483
2100              85.608483
2030              85.608483
2095              85.608483
2090              85.608483
2025              85.608483
2080              85.608483
2085              85.608483
2050              85.608483
2035              85.608483
2040              85.608483
2065              85.608483
2070              85.608483
2075              85.608483
2060              85.608483
2055              85.608483
2020              85.608483
1970              79.774206
1997              79.448245
1991              79.172927
1992              78.863474
1993              78.793526
1996              78.509814
1994              78.326548
1998              76.241519
1975              75.572250
1980              75.064143
1985              74.735664
2014              68.162449
1999              66.749486
2001              65.442845
2002              65.248108
1990              65.192149
2004              63.958534
2003              63.525133
2015              63.330675
1995              63.245898
2008              62.399239
2007              61.592031
2013              61.525720
2006              60.741456
2009              60.238945
2011              59.146626
2012              58.796323
2000              50.567004
2005              48.487570
2010              32.166030
Indicator Code     0.000000
Indicator Name     0.000000
Country Code       0.000000
Country Name       0.000000
dtype: float64

Maintenant que les lignes full NaN sont supprimées j'observe que les annèes cette répartition :
* colonne quasiment vides (>85%) : 2017, 2016, 1971 à 1986
* Colonnes avec beaucoup de NaN mais encore acceptable (50-85%) : 2045 à 2085, 1970 à 1998, 2000 à 2014
* Colonne plutot complètes (<50%) : 2025, 2010



#EDSFootNote

In [9]:
EDSFootNote.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643638 entries, 0 to 643637
Data columns (total 5 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   CountryCode  643638 non-null  object 
 1   SeriesCode   643638 non-null  object 
 2   Year         643638 non-null  object 
 3   DESCRIPTION  643638 non-null  object 
 4   Unnamed: 4   0 non-null       float64
dtypes: float64(1), object(4)
memory usage: 24.6+ MB


==> EDSFootNote est composé de 5 colonne :
* CountryCode ===> Code à 3 lettre
* SeriesCode ==> Chaine de caratère
* Year ==> Chaine de caratère également
* Description ==> Chaine de caratère
* Unnamed : 4 ==> ???

et de 643638 entrées

In [10]:
EDSFootNote_copy = EDSFootNote.copy()

In [11]:
EDSFootNote.isnull().sum()

CountryCode         0
SeriesCode          0
Year                0
DESCRIPTION         0
Unnamed: 4     643638
dtype: int64

La totalité des cases sont remplis pour l'ensemble des colonnes mis à part Unnamed : 4 qui est completement vide ==> Suppression de cette colonne

In [12]:
EDSFootNote = EDSFootNote.drop(columns=["Unnamed: 4"])
EDSFootNote.head()

CountryCode      SeriesCode    Year          DESCRIPTION
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.

Doublons

In [13]:
EDSFootNote.loc[EDSFootNote.duplicated(keep=False), :]

Empty DataFrame
Columns: [CountryCode, SeriesCode, Year, DESCRIPTION]
Index: []

Le dataset est "propre" et uniquement sur de la donnée qualitative.

#EDSSeries

In [15]:
EDSSeries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Series Code                          3665 non-null   object 
 1   Topic                                3665 non-null   object 
 2   Indicator Name                       3665 non-null   object 
 3   Short definition                     2156 non-null   object 
 4   Long definition                      3665 non-null   object 
 5   Unit of measure                      0 non-null      float64
 6   Periodicity                          99 non-null     object 
 7   Base Period                          314 non-null    object 
 8   Other notes                          552 non-null    object 
 9   Aggregation method                   47 non-null     object 
 10  Limitations and exceptions           14 non-null     object 
 11  Notes from original source    

Ce Dataset est composé de 21 colonnes et de 3665 lignes.

In [18]:
EDSSeries.isnull().mean()*100

Series Code                              0.000000
Topic                                    0.000000
Indicator Name                           0.000000
Short definition                        41.173261
Long definition                          0.000000
Unit of measure                        100.000000
Periodicity                             97.298772
Base Period                             91.432469
Other notes                             84.938608
Aggregation method                      98.717599
Limitations and exceptions              99.618008
Notes from original source             100.000000
General comments                        99.618008
Source                                   0.000000
Statistical concept and methodology     99.372442
Development relevance                   99.918145
Related source links                    94.133697
Other web links                        100.000000
Related indicators                     100.000000
License Type                           100.000000
Unnamed: 20                            100.000000
dtype: float64

J'ai 6 colonnes complétement vide et 4 colonnes completement rempli. Le reste est plutot vide (>90%)

In [19]:
EDSSeries_copy = EDSSeries.copy()

In [22]:
EDSSeries = EDSSeries.drop(columns=["Unit of measure", "Other web links", "Related indicators", "License Type", "Unnamed: 20"])
EDSSeries.head()

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition Periodicity Base Period  \
0  Percentage of female population age 15-19 with...         NaN         NaN   
1  Percentage of population age 15-19 with no edu...         NaN         NaN   
2  Percentage of female population age 15+ with n...         NaN         NaN   
3  Percentage of population age 15+ with no educa...         NaN         NaN   
4  Percentage of female population age 20-24 with...         NaN         NaN   

  Other notes Aggregation method Limitations and exceptions  \
0         NaN                NaN                        NaN   
1         NaN                NaN                        NaN   
2         NaN                NaN                        NaN   
3         NaN                NaN                        NaN   
4         NaN                NaN                        NaN   

   Notes from original source General comments  \
0                         NaN              NaN   
1                         NaN              NaN   
2                         NaN              NaN   
3                         NaN              NaN   
4                         NaN              NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN

#EDSCountry

In [24]:
EDSCountry.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

==> Dataset de 32 colonnes pour 241 lignes. Principalement du type Object sauf les colonnes 11, 28 et 29 en float

In [25]:
EDSCountry.isnull().mean()*100

Country Code                                           0.000000
Short Name                                             0.000000
Table Name                                             0.000000
Long Name                                              0.000000
2-alpha code                                           1.244813
Currency Unit                                         10.788382
Special Notes                                         39.834025
Region                                                11.203320
Income Group                                          11.203320
WB-2 code                                              0.414938
National accounts base year                           14.937759
National accounts reference year                      86.721992
SNA price valuation                                   18.257261
Lending category                                      40.248963
Other groups                                          75.933610
System of National Accounts                           10.788382
Alternative conversion factor                         80.497925
PPP survey year                                       39.834025
Balance of Payments Manual in use                     24.896266
External debt Reporting status                        48.547718
System of trade                                       17.012448
Government Accounting concept                         33.195021
IMF data dissemination standard                       24.896266
Latest population census                              11.618257
Latest household survey                               41.493776
Source of most recent Income and expenditure data     33.609959
Vital registration complete                           53.941909
Latest agricultural census                            41.078838
Latest industrial data                                55.601660
Latest trade data                                     23.236515
Latest water withdrawal data                          25.726141
Unnamed: 31                                          100.000000
dtype: float64

In [26]:
EDSCountry_copy = EDSCountry.copy()

In [28]:
EDSCountry = EDSCountry.drop(columns=[ "Unnamed: 31"])

KeyError: "['Unnamed: 31'] not found in axis"

In [30]:
EDSCountry.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 31 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

#EDSCountrySeries

In [32]:
EDSCountrySeries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB


==> Ce dataset est composé de 4 colonnes et de 613 entrées. l'ensemble des colonnes est au format Object

In [33]:
EDSCountrySeries.isnull().mean()*100

CountryCode      0.0
SeriesCode       0.0
DESCRIPTION      0.0
Unnamed: 3     100.0
dtype: float64

==> L'ensemble des colonnes sont remplis à 100% à part Unnamed qui est complétement vide

In [34]:
EDSCountrySeries_copy = EDSCountrySeries.copy()

In [35]:
EDSCountrySeries = EDSCountrySeries.drop(columns=["Unnamed: 3"])
EDSCountrySeries

CountryCode         SeriesCode  \
0           ABW        SP.POP.TOTL   
1           ABW        SP.POP.GROW   
2           AFG        SP.POP.GROW   
3           AFG  NY.GDP.PCAP.PP.CD   
4           AFG        SP.POP.TOTL   
..          ...                ...   
608         ZAF        SP.POP.GROW   
609         ZMB        SP.POP.GROW   
610         ZMB        SP.POP.TOTL   
611         ZWE        SP.POP.TOTL   
612         ZWE        SP.POP.GROW   

                                           DESCRIPTION  
0    Data sources : United Nations World Population...  
1    Data sources: United Nations World Population ...  
2    Data sources: United Nations World Population ...  
3                   Estimates are based on regression.  
4    Data sources : United Nations World Population...  
..                                                 ...  
608  Data sources : Statistics South Africa, United...  
609  Data sources: United Nations World Population ...  
610  Data sources : United Nations World Population...  
611  Data sources : United Nations World Population...  
612  Data sources: United Nations World Population ...  

[613 rows x 3 columns]

#3°/ Compréhension des liens des dataset et choix d'analyse

Etudier le rapport entre chaque dataset

In [46]:
EDSData = EDSData.rename(columns={"Country Code": "CountryCode"})
EDSCountry = EDSCountry.rename(columns={"Country Code": "CountryCode"})

In [47]:
datasets = {
    "EDSData": EDSData,
    "EDSFootNote": EDSFootNote,
    "EDSSeries": EDSSeries,
    "EDSCountry": EDSCountry,
    "EDSCountrySeries": EDSCountrySeries
}

for name1, df1 in datasets.items():
    for name2, df2 in datasets.items():
        if name1 != name2:
            common_columns = df1.columns.intersection(df2.columns)
            if common_columns.empty == False:
              print(f"{name1} n {name2} : {common_columns}")


EDSData n EDSFootNote : Index(['CountryCode'], dtype='object')
EDSData n EDSSeries : Index(['Indicator Name'], dtype='object')
EDSData n EDSCountry : Index(['CountryCode'], dtype='object')
EDSData n EDSCountrySeries : Index(['CountryCode'], dtype='object')
EDSFootNote n EDSData : Index(['CountryCode'], dtype='object')
EDSFootNote n EDSCountry : Index(['CountryCode'], dtype='object')
EDSFootNote n EDSCountrySeries : Index(['CountryCode', 'SeriesCode', 'DESCRIPTION'], dtype='object')
EDSSeries n EDSData : Index(['Indicator Name'], dtype='object')
EDSCountry n EDSData : Index(['CountryCode'], dtype='object')
EDSCountry n EDSFootNote : Index(['CountryCode'], dtype='object')
EDSCountry n EDSCountrySeries : Index(['CountryCode'], dtype='object')
EDSCountrySeries n EDSData : Index(['CountryCode'], dtype='object')
EDSCountrySeries n EDSFootNote : Index(['CountryCode', 'SeriesCode', 'DESCRIPTION'], dtype='object')
EDSCountrySeries n EDSCountry : Index(['CountryCode'], dtype='object')
